velox/type/TimestampConversion.cpp

```diff
@@ -758,28 +802,38 @@ Expected<Timestamp> fromTimestampString(const char* str, size_t len) {
   return resultTimestamp;
 }
 
-Expected<std::pair<Timestamp, int64_t>> fromTimestampWithTimezoneString(
+Expected<std::pair<Timestamp, int16_t>> fromTimestampWithTimezoneString(
     const char* str,
-    size_t len) {
-  size_t pos;
+    size_t len,
+    TimestampParseMode parseMode) {
+  size_t pos = 0;
   Timestamp resultTimestamp;
 
-  if (!tryParseTimestampString(str, len, pos, resultTimestamp)) {
+  if (!tryParseTimestampString(str, len, pos, resultTimestamp, parseMode)) {
     return folly::makeUnexpected(parserError(str, len));
   }
 
-  int64_t timezoneID = -1;
+  int16_t timezoneID = -1;
 
   if (pos < len && characterIsSpace(str[pos])) {
     pos++;
   }
 
   // If there is anything left to parse, it must be a timezone definition.
   if (pos < len) {
+    if (parseMode == TimestampParseMode::kIso8601) {
+      // Only +HH:MM and -HH:MM are supported. Minutes, seconds, etc. in the
+      // offset are optional.
```

Comment:
you can't have seconds in the timezone offset; only hours and minutes

Reply:
Presto thinks otherwise:

```
presto:di> select from_iso8601_timestamp('T01:01:01+05:30:00.000');
             _col0
--------------------------------
 1970-01-01 01:01:01.000 +05:30
(1 row)
```